services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
+	return paths
+}
+
+func verifyCommitPathsExist(fs billy.Filesystem, paths []string) error {
+	for _, path := range paths {
+		_, err := fs.Stat(path)
+		if err != nil {
+			return fmt.Errorf(`error verifying commit path exists. path: %s, error: %v
+			
+			directory tree: %s`, path, err, strings.Join(dumpFilesystem(fs), "\n"))
+		}
+	}
+	return nil
+}
+
+func verifyCommitPathsDontExist(fs billy.Filesystem, paths []string) error {
+	for _, path := range paths {
+		_, err := fs.Stat(path)
+		if err == nil {
+			return fmt.Errorf(`error verifying commit path doesn't exist. path: %s, error expected but none was raised
+			
+			directory tree: %s`, path, strings.Join(dumpFilesystem(fs), "\n"))
+		}
+	}
+	return nil
+}
+
+func verifyConsistency(fs billy.Filesystem) error {
+	type ApplicationDirectoryContent struct {
+		application    string
+		sourceCommitID string
+	}
+	applications := make([]ApplicationDirectoryContent, 0)
+
+	{
+		applicationsDir, err := fs.ReadDir("applications")
+		if err != nil {
+			return fmt.Errorf("could not open the applications directory: %w", err)
+		}
+
+		for _, applicationDir := range applicationsDir {
+			releasesDir, err := fs.ReadDir(fs.Join("applications", applicationDir.Name(), "releases"))
+			if err != nil {
+				return fmt.Errorf("could not open the releases directory: %w", err)
+			}
+			for _, releaseDir := range releasesDir {
+				commitIDFile, err := util.ReadFile(fs, fs.Join("applications", applicationDir.Name(), "releases", releaseDir.Name(), "source_commit_id"))
+
+				if err != nil {
+					return fmt.Errorf("could not read the source commit ID file: %w", err)
+				}
+
+				sourceCommitID := string(commitIDFile)
+				if valid.SHA1CommitID(sourceCommitID) {
+					applications = append(applications, ApplicationDirectoryContent{
+						application:    applicationDir.Name(),
+						sourceCommitID: sourceCommitID,
+					})
+				}
+			}
+		}
+	}
+
+	type CommitDirectoryContent struct {
+		application    string
+		sourceCommitID string
+	}
+	commits := make([]CommitDirectoryContent, 0)
+
+	{
+		commitsDir1, err := fs.ReadDir("commits")
+		if err != nil {
+			return fmt.Errorf("could not open the commits directory: %w", err)
+		}
+
+		for _, commitDir1 := range commitsDir1 {
+			commitsDir2, err := fs.ReadDir(fs.Join("commits", commitDir1.Name()))
+			if err != nil {
+				return fmt.Errorf("could not open the commit directory 1")
+			}
+
+			for _, commitDir2 := range commitsDir2 {
+				applicationsDir, err := fs.ReadDir(fs.Join("commits", commitDir1.Name(), commitDir2.Name(), "applications"))
+				if err != nil {
+					return fmt.Errorf("could not open the applications directory in the commits tree: %w", err)
+				}
+
+				for _, applicationDir := range applicationsDir {
+					commits = append(commits, CommitDirectoryContent{
+						application:    applicationDir.Name(),
+						sourceCommitID: commitDir1.Name() + commitDir2.Name(),
+					})
+				}
+			}
+		}
+	}
```

Comment:
nitpick: maybe extract this into its own function, e.g. `getAllCommits`

Reply:
Refactored into functions, should be okay now.